clang/test/dpct/asm/prefetch.cu

```diff
@@ -0,0 +1,91 @@
+
+Unsupported syntax:
+-------------------
+prefetch.local.level
+prefetch.global.level::eviction_priority [a];   // prefetch to data cache
+prefetchu.L1 [a];                               // prefetch to uniform cache
+prefetch{.tensormap_space}.tensormap [a];       // prefetch the tensormap
+
+.level =                    { .L1, .L2 };
+.level::eviction_priority = { .L2::evict_last, .L2::evict_normal };
+.tensormap_space =          { .const, .param };
+*/
+
+__global__ void prefetch(int *arr) {
+  // CHECK: sycl::ext::oneapi::experimental::prefetch(arr, sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::prefetch_hint_L1});
+  asm volatile ("prefetch.L1 [%0];" : : "l"(arr));
+  // CHECK: sycl::ext::oneapi::experimental::prefetch(arr, sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::prefetch_hint_L2});
+  asm volatile ("prefetch.L2 [%0];" : : "l"(arr));
+#ifndef BUILD_TEST
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.L2::evict_last [%0];" : : "l"(arr));
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.L2::evict_normal [%0];" : : "l"(arr));
+#endif // BUILD_TEST
+
+  // CHECK: sycl::ext::oneapi::experimental::prefetch(arr, sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::prefetch_hint_L1});
+  asm volatile ("prefetch.global.L1 [%0];" : : "l"(arr));
+  // CHECK: sycl::ext::oneapi::experimental::prefetch(arr, sycl::ext::oneapi::experimental::properties{sycl::ext::oneapi::experimental::prefetch_hint_L2});
+  asm volatile ("prefetch.global.L2 [%0];" : : "l"(arr));
+#ifndef BUILD_TEST
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.global.L2::evict_last [%0];" : : "l"(arr));
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.global.L2::evict_normal [%0];" : : "l"(arr));
+
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.local.L1 [%0];" : : "l"(arr));
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.local.L2 [%0];" : : "l"(arr));
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.local.L2::evict_last [%0];" : : "l"(arr));
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1053:{{.*}} Migration of device assembly code is not supported.
+  // CHECK-NEXT: */
+  asm volatile ("prefetch.local.L2::evict_normal [%0];" : : "l"(arr));
```

Comment:
It is a invalid PTX instruction in CUDA side, according to https://docs.nvidia.com/cuda/parallel-thread-execution/#data-movement-and-conversion-instructions-prefetch-prefetchu

Reply:
Removed now.
Initially it is added to show how SYCLomatic handles invalid ASM syntax expr